pcsx2/x86/microVU_Macro.inl

```diff
@@ -82,29 +96,26 @@ void endMacroOp(int mode)
 	{
 		xMOVSS(ptr32[&vu0Regs.VI[REG_Q].UL], xmmPQ);
 	}
-	if (mode & 0x10) // Status/Mac Flags were Updated
-	{
-		// Normalize
-		mVUallocSFLAGc(eax, gprF0, 0);
-		xMOV(ptr32[&vu0Regs.VI[REG_STATUS_FLAG].UL], eax);
-	}
 
 	microVU0.regAlloc->flushAll();
 	_clearNeededCOP2Regs();
 
-	if (mode & 0x10) // Update VU0 Status/Mac instances after flush to avoid corrupting anything
+	if (mode & 0x10)
 	{
-		int t0reg = _allocTempXMMreg(XMMT_INT, -1);
-		mVUallocSFLAGd(&vu0Regs.VI[REG_STATUS_FLAG].UL);
-		xMOVDZX(xRegisterSSE(t0reg), eax);
-		xSHUF.PS(xRegisterSSE(t0reg), xRegisterSSE(t0reg), 0);
-		xMOVAPS(ptr128[&microVU0.regs().micro_statusflags], xRegisterSSE(t0reg));
-
-		xMOVDZX(xRegisterSSE(t0reg), ptr32[&vu0Regs.VI[REG_MAC_FLAG].UL]);
-		xSHUF.PS(xRegisterSSE(t0reg), xRegisterSSE(t0reg), 0);
-		xMOVAPS(ptr128[&microVU0.regs().micro_macflags], xRegisterSSE(t0reg));
-		_freeXMMreg(t0reg);
+		if (!CHECK_COP2_FLAGHACK || g_pCurInstInfo->info & EEINST_COP2_NORMALIZE_STATUS_FLAG)
+		{
+			// Normalize
+			mVUallocSFLAGc(eax, gprF0, 0);
+			xMOV(ptr32[&vu0Regs.VI[REG_STATUS_FLAG].UL], eax);
+		}
+		else
+		{
+			// backup denormalized flags for the next instruction
+			// this is fine, because we'll normalize them again before this reg is accessed
+			xMOV(gprF0, ptr32[&vuRegs->VI[REG_STATUS_FLAG].UL]);
```

Comment:
Operands are backwards (this is what breaks the BIOS) but correcting it re-breaks Tekken Tag

Reply:
seems to want status flags to be written on line 63
if (mode & 0x10 && (!CHECK_COP2_FLAGHACK || g_pCurInstInfo->info & EEINST_COP2_STATUS_FLAG)) // Update Status Flag

removing the EEINST_COP2_STATUS_FLAG check fixes it.

Reply:
Just an additional, I set it to always update those flags on the instructions that want it, and the performance was the same on x86 (maybe margin of error better), is it a huge problem on ARM?

Reply:
I had a ponder about this when I went for a nap. I guess you *could* do what you are doing now, but you'd have to scan forward on the block, see if it reads the status register, followed by an AND with a value including any bits of 0xFC0, if that's true, enable status/mac updates for that entire block, that will probably be more than enough for it.